torch/functional.py

```diff
@@ -102,21 +102,37 @@ def broadcast_shapes(*shapes):
     """
     # This wrapper exists to support variadic args.
     # TODO Movie this to C++ once the jit has better support for torch.Size.
-    with torch.no_grad():
-        scalar = torch.zeros((), device="cpu")
-
-        for shape in [*shapes]:
-            if isinstance(shape, int) and shape < 0:
-                raise RuntimeError(rf"Trying to create tensor with negative dimension {shape}: [{shape}]")
-            if isinstance(shape, tuple):
-                for ele in shape:
-                    if ele < 0:
-                        raise RuntimeError(rf"Trying to create tensor with negative dimension {ele}: [{ele}]")
-
-        tensors = [scalar.expand(shape) for shape in shapes]
-        tensors = broadcast_tensors(*tensors)
-        return tensors[0].shape
-
+    for shape in shapes:
```

Comment:
you don't need to make this check separately in the beginning, as you are going to be iterating over all shapes values later, and can check them there

Reply:
Yeah, agreed. I've updated it. Thanks :)